pkg/apis/internal/gs/state.go

```diff
@@ -0,0 +1,28 @@
+package gs
+
+import (
+	"go.anx.io/go-anxcloud/pkg/api/types"
+)
+
+// State represents the state object available on GS resources
+type State struct {
+	// programatically usable enum value
+	ID string `json:"id"`
+
+	// human readable status text
+	Text string `json:"text"`
+
+	Type int `json:"type"`
```

Comment:
we could probably use this field to find the state category(?) (OK, Pending, Error) for all generic services